helpers.R

```diff
@@ -372,31 +371,31 @@ filter_dropdown_varlist = function(filter_set, vartype='stream'){
     return(vars_display_subset)
 }
 
-# df=data3; agg_selection=agg3; which_dataset='pchem'; conc_flux_selection=conc_flux3
-# df=pchem3; agg_selection=agg3; which_dataset='pchem'; conc_flux_selection=conc_flux3
+# df=data3; agg_selection=agg; which_dataset='pchem'; conc_flux_selection=conc_flux
+# df=pchem3; agg_selection=agg; which_dataset='pchem'; conc_flux_selection=conc_flux
 ms_aggregate = function(df, agg_selection, which_dataset,
     conc_flux_selection=NULL){
 
     #agg_selection is a user input object, e.g. input$AGG3
-    #which_dataset is one of 'grab', 'q', 'p', 'pchem'
+    #which_dataset is one of 'chem', 'q', 'p', 'pchem'
     #conc_flux_selection must be supplied as e.g. input$CONC_FLUX3 if
-    #which_dataset is 'grab' or 'pchem'
+    #which_dataset is 'chem' or 'pchem'
 
-    if(! which_dataset %in% c('grab', 'q', 'p', 'pchem')){
-        stop("which_dataset must be one of 'grab', 'q', 'p', 'pchem'")
+    if(! which_dataset %in% c('chem', 'q', 'p', 'pchem')){
+        stop("which_dataset must be one of 'chem', 'q', 'p', 'pchem'")
     }
 
-    if(which_dataset %in% c('grab', 'pchem') && is.null(conc_flux_selection)){
+    if(which_dataset %in% c('chem', 'pchem') && is.null(conc_flux_selection)){
         stop(paste0("conc_flux_selection must be supplied when which_dataset",
-            "is 'grab' or'pchem'"))
+            "is 'chem' or'pchem'"))
     }
 
     if(nrow(df) == 0) return(df)
     if(agg_selection == 'Instantaneous') return(df)
     if(agg_selection == 'Daily' && which_dataset == 'pchem') return(df)
 
-    agg_period = switch(agg_selection, 'Daily'='day', 'Monthly'='month',
-        'Yearly'='year')
+    agg_period = switch(agg_selection,
```

Comment:
Where does the switch come from? Is it a base thing, wonder if case_when and/or lookup tibble is better?

Reply:
It is base. I'll use case_when from now on, in keeping with our tidy goals.